mozdownload/scraper.py

```diff
@@ -332,10 +335,15 @@ def show_matching_builds(self, builds):
 class DailyScraper(Scraper):
     """Class to download a daily build from the Mozilla server."""
 
-    def __init__(self, branch='mozilla-central', build_id=None, date=None,
+    def __init__(self, branch=None, build_id=None, date=None,
                  build_number=None, revision=None, *args, **kwargs):
         """Create an instance of the daily scraper."""
-        self.branch = branch
+        if not branch:
+            application = kwargs.get('application', None)
```

Comment:
@whimboo Moving the code below Scraper.__init__() is not working since it is using self.branch which is not set.